src/systemctl/systemctl.c

```diff
@@ -1,8965 +1,874 @@
-        if (IN_SET(r, -ETIMEDOUT, -ECONNRESET) && streq(method, "Reexecute"))
-                r = 0;
-
-        if (r < 0 && arg_action == ACTION_SYSTEMCTL)
-                return log_error_errno(r, "Failed to reload daemon: %s", bus_error_message(&error, r));
-
-        /* Note that for the legacy commands (i.e. those with action != ACTION_SYSTEMCTL) we support fallbacks to the
-         * old ways of doing things, hence don't log any error in that case here. */
-
-        return r < 0 ? r : 0;
-}
-
-static int trivial_method(int argc, char *argv[], void *userdata) {
-        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
-        const char *method;
-        sd_bus *bus;
-        int r;
-
-        if (arg_dry_run)
-                return 0;
-
-        r = acquire_bus(BUS_MANAGER, &bus);
-        if (r < 0)
-                return r;
-
-        polkit_agent_open_maybe();
-
-        method =
-                streq(argv[0], "clear-jobs")    ||
-                streq(argv[0], "cancel")        ? "ClearJobs" :
-                streq(argv[0], "reset-failed")  ? "ResetFailed" :
-                streq(argv[0], "halt")          ? "Halt" :
-                streq(argv[0], "reboot")        ? "Reboot" :
-                streq(argv[0], "kexec")         ? "KExec" :
-                streq(argv[0], "exit")          ? "Exit" :
-                             /* poweroff */       "PowerOff";
-
-        r = bus_call_method(bus, bus_systemd_mgr, method, &error, NULL, NULL);
-        if (r < 0 && arg_action == ACTION_SYSTEMCTL)
-                return log_error_errno(r, "Failed to execute operation: %s", bus_error_message(&error, r));
-
-        /* Note that for the legacy commands (i.e. those with action != ACTION_SYSTEMCTL) we support fallbacks to the
-         * old ways of doing things, hence don't log any error in that case here. */
-
-        return r < 0 ? r : 0;
-}
-
-static int reset_failed(int argc, char *argv[], void *userdata) {
-        _cleanup_strv_free_ char **names = NULL;
-        sd_bus *bus;
-        char **name;
-        int r, q;
-
-        if (argc <= 1)
-                return trivial_method(argc, argv, userdata);
-
-        r = acquire_bus(BUS_MANAGER, &bus);
-        if (r < 0)
-                return r;
-
-        polkit_agent_open_maybe();
-
-        r = expand_names(bus, strv_skip(argv, 1), NULL, &names, NULL);
-        if (r < 0)
-                return log_error_errno(r, "Failed to expand names: %m");
-
-        STRV_FOREACH(name, names) {
-                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
-
-                q = bus_call_method(bus, bus_systemd_mgr, "ResetFailedUnit", &error, NULL, "s", *name);
-                if (q < 0) {
-                        log_error_errno(q, "Failed to reset failed state of unit %s: %s", *name, bus_error_message(&error, q));
-                        if (r == 0)
-                                r = q;

```

Comment:
The changes to this function strike me as more than "No change of behaviour, just reshuffling of some code.", I expected to find an essentially mechanical change when skimming the diff and that doesn't look to be the case in this function.

Maybe the changes are trivial and obviously correct to someone spending more time on a proper review, but I had hoped a quick LGTM would be possible from a skim.  I'd prefer the spurious non-mechanical changes be in separate commits.  This particular commit should be strictly limited to the mechanical splitting.

Reply:
hmm so github sucks with showing the diff for this. git actually too. the diff is smaller than it appears, we really just remove stuff from systemctl.c because it its placed in those other .c files.

The git diff on the cmdline becomes more digestable if you run it with `--minimal`, since it then will try harder to match up the code fragments and then realizes that the changes are much simpler than they appear otherwise.

Reply:
(unfortunately the github UI doesn't have an equivalent for `--minimal`, probably because they don't want users to spend arbitrary amounts of thei CPU time on diffing humoungous files)

Reply:
Indeed the git add/remove rendering on this was very unfortunate as a result

Reply:
I see, indeed this is a github failure.